bpf/bpf_overlay.c

```diff
@@ -333,6 +339,51 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 					      CTX_ACT_DROP, METRIC_INGRESS);
 	return ret;
 }
+
+#ifdef ENABLE_VTEP
+/*
+ * ARP responder for ARP requests from VTEP
+ * Respond to remote VTEP endpoint with cilium_vxlan MAC
+ */
+__section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_ARP)
+int tail_handle_arp(struct __ctx_buff *ctx)
+{
+	union macaddr mac = NODE_MAC;
+	union macaddr smac;
+	__be32 sip;
+	__be32 tip;
+	__u32 monitor = 0;
+	int i;
+	int ret;
+	struct remote_endpoint_info *info;
+	struct bpf_tunnel_key key = {};
+
+	if (unlikely(ctx_get_tunnel_key(ctx, &key, sizeof(key), 0) < 0))
+		return send_drop_notify_error(ctx, 0, DROP_NO_TUNNEL_KEY, CTX_ACT_DROP,
+										METRIC_INGRESS);
+
+	if (!arp_validate(ctx, &mac, &smac, &sip, &tip))
+		return CTX_ACT_OK;
+	if (!__lookup_ip4_endpoint(tip))
+		return CTX_ACT_OK;
+	info = lookup_ip4_remote_endpoint(sip);
+	if (!info)
+		return CTX_ACT_OK;
+
+	ret = arp_prepare_response(ctx, &mac, tip, &smac, sip);
+	if (unlikely(ret != 0))
+		return send_drop_notify_error(ctx, 0, ret, CTX_ACT_DROP, METRIC_EGRESS);
+
+	for (i = 0; i < VTEP_NUMS; i++) {
+		if (info->tunnel_endpoint == VTEP_ENDPOINT[i])
+			return __encap_and_redirect_with_nodeid(ctx, info->tunnel_endpoint,
+									info->sec_label, monitor);
+	}
+
+	return send_drop_notify_error(ctx, 0, DROP_INVALID, CTX_ACT_DROP, METRIC_EGRESS);
```

Comment:
`DROP_INVALID` is already overloaded to handle a bunch of other "packet is bad" type error messages, I think we can be a bit more concrete about the failure here by indicating that it's a problem reaching the L3 peer, so your suggestion makes sense to me:

```suggestion
	return send_drop_notify_error(ctx, 0, DROP_UNKNOWN_L3, CTX_ACT_DROP, METRIC_EGRESS);
```